master/internal/user/service.go

```diff
@@ -108,20 +144,34 @@ func (s *Service) UserAndSessionFromRequest(
 	return UserByToken(token, s.extConfig)
 }
 
+// getAuthLevel returns what level of authentication a request needs.
+func (s *Service) getAuthLevel(c echo.Context) int {
+	switch {
+	case adminAuthPointsPattern.MatchString(c.Request().RequestURI):
+		return authAdmin
+	case unauthenticatedPointsPattern.MatchString(c.Path()):
+		return authNone
```

Comment:
Remove this. I have confirmed that the agents endpoint would be matched by `RequestURI`.

Reply:
looks like this is necessary to pass integration/e2e tests. Will investigate a bit more.